WordPress/Classes/Utility/CoreDataHelper.swift

```diff
@@ -189,7 +189,9 @@ extension CoreDataStack {
             if case .success = result {
                 self.saveContextAndWait(context)
             }
-            completion?(result)
+            DispatchQueue.main.async {
```

Comment:
Can we please require the caller to specify the queue as an argument rather than scheduling work on the main thread after each save?

Reply:
PR created [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/19959) 😄